### DISCUSSION TITLE
SQLite: Creating object with a nonexistent foriegn key does not throw Error

### DISCUSSION THREAD

**Discussion by petercinibulk** [sqlalchemy/sqlalchemy#10072] [Usage Questions] [answered]
### Describe the bug

When adding an object to the database with a foreign key to another table, I understand the documentation to say that it should throw an error if the foreign key does not exist. This does not happen.

I could understand it not throwing an error if the relationship was optional, but it is not in my use case.

If my understanding is incorrect, the docs should be updated to highlight this but I couldn't find anything in the docs.

### Optional link from https://docs.sqlalchemy.org which documents the behavior that is expected

https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html

### SQLAlchemy Version in Use

2.0.17

### DBAPI (i.e. the database driver)

psycopg2

### Database Vendor and Major Version

SQLite

### Python Version

3.8.16

### Operating system

Linux

### To Reproduce

```python
from sqlalchemy import ForeignKey, Integer, String
from sqlalchemy.orm import DeclarativeBase, relationship, mapped_column
import os

from sqlalchemy import create_engine
from sqlalchemy.orm import Session


class Base(DeclarativeBase):
    pass


class User(Base):
    __tablename__ = "user"
    id = mapped_column(Integer, primary_key=True)
    name = mapped_column(String)
    addresses = relationship("Address", back_populates="user")


class Address(Base):
    __tablename__ = "address"
    id = mapped_column(Integer, primary_key=True)
    user_id = mapped_column(Integer, ForeignKey("user.id"))
    user = relationship("User", back_populates="addresses")


db_path = f"./test.db"
sqlite_url = f"sqlite:///{db_path}"
engine = create_engine(sqlite_url, connect_args={"check_same_thread": False})

Base.metadata.create_all(engine)
with Session(engine) as session:
    address = Address(user_id=-1)
    session.add(address)
    session.commit()
    session.refresh(address)
    print(address.user) # Prints "None"

if os.path.exists(db_path):
    os.remove(db_path)
```

### Error

No error produced

**Comment by lelit** [comment, marked as answer]
On SQLite you must explicitly enable FKs, otherwise they are simply ignored, as explained [here](https://docs.sqlalchemy.org/en/20/dialects/sqlite.html#foreign-key-support).
